src/tools/idlc/src/plugin.c

```diff
@@ -64,6 +70,59 @@ static void *loadsym(void *handle, const char *symbol)
 #endif
 }
 
+#define SUBPROCESS_PIPE_MEMORY_LIMIT 1024 * 1024
+static int run_library_locator(const char *command, char **out_output, int *out_return_code) {
+  size_t output_size = 0;
+  size_t output_pt = 0;
+  char *output = NULL;
+  FILE *pipe;
+  int ret = 0;
+  int c;
+
+  if ((pipe = SUBPROCESS_POPEN (command, "r")) == NULL) {
+    // broken-pipe
+    return -1;
+  }
+
+  while ((c = fgetc(pipe)) != EOF) {
+    if (c == '\n' || c == '\r') {
+      break;
+    }
+
+    if (output_pt == output_size) {
+      output_size += 128;
+      if (output_size > SUBPROCESS_PIPE_MEMORY_LIMIT) {
+        free(output);
+        SUBPROCESS_PCLOSE (pipe);
+        return -1;
+      }
+
+      char* new = (char*) realloc(output, output_size);
+      if (!new) {
+        free(output);
+        SUBPROCESS_PCLOSE (pipe);
+        return -1;
+      }
+      output = new;
+    }
+
+    output[output_pt++] = (char) c;
+  }
+
+  if (output == NULL) {
+    // no data received
+  } else {
+    // ensure proper string termination (might be newline)
+    output[output_pt] = '\0';
+  }
+
+  *out_output = output;
+  *out_return_code = SUBPROCESS_PCLOSE (pipe);
```

Comment:
I'd just return `-1` if the exit code is not `0`. Also, just return `-1` if the there's no output. If no output is received, we cannot continue anyway. So, if `0` is returned from this function there must be `output`, otherwise, if the process return code is non-zero, free `output`(?) Much easier.